### PR TITLE
Fix: ZMUpdateEvent stub not work on XCFramework

### PR DESCRIPTION
**Pull request by billypchan**
### Issues

After using XCFramework, it is no longer allow stub `ZMUpdateEvent` `messageNonce` property. After remove that stub, some tests crashes

### Causes

`messageNonce` no longer allow  be marked @objc since it is in extension 
After using XCFramework, `ZMUpdateEvent` stub no longer works and the tests calls original code

### Solutions

~disable those tests and refactor them without stub later.~ Add simulator only guard in `updateEventParticipantsRemovedReason` to prevent accessing `ZMUpdateEvent.payload` stub

### CODE REVIEW

File: Source/Model/Message/ZMSystemMessage+ChildMessages.swift [hunk 1]
```diff
@@ -22,7 +22,15 @@ extension ZMSystemMessage {
     
     @objc
     static func updateEventParticipantsRemovedReason(_ updateEvent: ZMUpdateEvent) -> ZMParticipantsRemovedReason {
-            return updateEvent.participantsRemovedReason
+        
+        // fix: OCMock no longer work after XCFramework is used
```
Comment (typfel) [inline Source/Model/Message/ZMSystemMessage+ChildMessages.swift:26]: Create a ticket for removing the usage of OCMock and link it here.

Comment (typfel) [reply Source/Model/Message/ZMSystemMessage+ChildMessages.swift:26]: Or remove the OCMock usage in this PR doesn't look like it's that much work.

Comment (billypchan) [reply Source/Model/Message/ZMSystemMessage+ChildMessages.swift:26]: Other tests are still depends on OCMock, e.g. `testThatItSavesTheConversationTitleInConversationNameChangeSystemMessage` still calls objc code. I would prefer keep this minimal change and remove ocmock later.

Comment (billypchan) [reply Source/Model/Message/ZMSystemMessage+ChildMessages.swift:26]: > Create a ticket for removing the usage of OCMock and link it here.

https://wearezeta.atlassian.net/browse/SQPIT-601

Comment (typfel) [inline Source/Model/Message/ZMSystemMessage+ChildMessages.swift:26]:
```suggestion
        // FUTUREWORK: OCMock no longer work after XCFramework is used. We should remove the OCMock dependency in the tests SQPIT-601
```

**Review by typfel — APPROVED** [review]

**Review by David-Henner — APPROVED** [review]

**Pull request merged**